select_samples_grain_size.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+from scipy.ndimage import center_of_mass
+
+
+def select_samples_grain_size(mask, bins, num_samples=1, setting='highest'):
```

Comment:
Here we can use type annotations. For example:

```python
def select_samples_grain_size(mask: np.ndarray, bins: int, num_samples: int = 1, setting: str = "highest") -> list[list[int]]:
```

Further, we want to use an enum for the setting (there are only two options!):

so --
 
```python
from enum import Enum
...

class GrainSizeSetting(str, Enum):
	HIGHEST = "highest"
	LOWEST = "lowest"

# and in the def:
 ... setting: GrainSizeSetting = GrainSizeSetting.HIGHEST):
 
 ...
 
 if setting == GrainSizeSetting.HIGHEST:
 	...
 elif setting == GrainSizeSetting.LOWEST:
 	...
 else:
 	...
 ```